static/js/components/widgets/SelectField.tsx

```diff
@@ -1,36 +1,58 @@
-import React, { useCallback, ChangeEvent } from "react"
+import React, { useCallback, ChangeEvent, useState } from "react"
 import Select from "react-select"
-import AsyncSelect from "react-select/async"
 import { isNil } from "ramda"
+import { AsyncPaginate, LoadOptions } from "react-select-async-paginate"
+import AsyncSelect from "react-select/async"
+import { isFeatureEnabled } from "../../util/features"
 
 export interface Option {
   label: string
   value: string
 }
 
+export interface Additional {
+  callback?: (options: Option[]) => void
+}
+
 interface Props {
   name: string
-  value: null | undefined | string | string[]
+  classNamePrefix?: string
+  value?: null | undefined | string | string[]
   onChange: (event: ChangeEvent<HTMLSelectElement>) => void
   multiple?: boolean
   options: Array<string | Option>
-  loadOptions?: (s: string, cb: (options: Option[]) => void) => void
+  loadOptions?: LoadOptions<Option, Option[], Additional | undefined>
   placeholder?: string
   defaultOptions?: Option[]
+  cacheUniques?: ReadonlyArray<any>
```

Comment:
`AsyncPaginate` caches search results internally. This prop is a set of values used to identify a cache inside `AsyncPaginate` component. This prop is useful when two SelectFields are related. For example, resource collection selects a website in one select and the resource in the second select field. When the website select changes, the resource select needs to reset/change its cache.